pkg/api/steve/catalog/content.go

```diff
@@ -108,10 +108,13 @@ func (i *contentDownload) serveIcon(apiContext *types.APIRequest, rw http.Respon
 	if err != nil {
 		return err
 	}
-	defer chart.Close()
+	// if no chart object is provided just return an emtpy response
```

Comment:
The first commit message has wrong content.

Reply:
Re-organized and separated all commits.

Reply:
Reorganized and separated all commits.